backup/crypto.go

```diff
@@ -0,0 +1,80 @@
+package backup
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"	
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+func encryptFile(source, dest string, key []byte) error {
+	fileContent, err := ioutil.ReadFile(source)
+	if err != nil {
+		return err
+	}
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return err
+	}
+
+	nonce := make([]byte, 12)
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		return err
+	}
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return err
+	}
+
+	encryptedContent := aesgcm.Seal(nonce, nonce, fileContent, nil)
```

Comment:
Does using nonce as dst make it so encryptedContent is nonce + ciphertex or just ciphertext? I see you are not reusing nonce in this func (in WriteFile), however according to docs I think that's what you want. :thinking: